### PR TITLE
Update tendermint version

### PR DESCRIPTION
**Pull request by Olshansk**
Update the tendermint fork version.

## Testing

Validated use LocalNet instructions at [doc/guides/localnet.md](https://github.com/pokt-network/pocket-core/blob/staging/doc/guides/localnet.md)

## Changes

- Update Tendermint to Go 1.18 + minor fixes: [tendermint/pull/74](https://github.com/pokt-network/tendermint/pull/74)
- Prevent peers who errored being added to the peer_set: [tendermint/pull/75](https://github.com/pokt-network/tendermint/pull/75)

### CODE REVIEW

**Review by msmania — DISMISSED** [review]
Confirmed on localnet.  Looks good!

I can see your struggle on go.mod stuff in the tendermint's commit history :)

**Comment by Olshansk** [comment]
> I can see your struggle on go.mod stuff in the tendermint's commit history :)

Haha, yea... Not a fun thing but not something we'll need to worry about in v1.

I'm actually going to hold of on merging this one in until @aqt01 launches a few nodes on TestNet to see how a heterogeneous (w.r.t to Tendermint) network behaves.

Screenshot 2023-04-07 at 10 56 56 AM

**Comment by fredteumer** [comment]
@Olshansk 

is this going to be deployed on mainnet or testnet?

also do you want the node to be spun from no state/genesis or from an existing state?

**Comment by Olshansk** [comment]
> @Olshansk
> is this going to be deployed on mainnet or testnet?
> also do you want the node to be spun from no state/genesis or from an existing state?

tl;dr I'm going to merge this into main and tag the commit as beta. We, as PNI, should deploy it on MainNet both from genesis and from the latest snapshot you have.

--- Extra optional details --- 

@fredteumer This has already been deployed on both TestNet and MainNet per the updates in the [Tendermint testing in the release guide](https://www.notion.so/pocketnetwork/Testing-Tendermint-fbcd3abbed5b4d1ea470a964e2e8b014?pvs=4).

@aqt01 and @StevenMartinez94 are maintaining it on TestNet

@msmania are deploying it on MainNet

The conversation is ongoing in the public discord channel [here](https://discord.com/channels/553741558869131266/1090451384358420520).

**Comment by Olshansk** [comment]
We've been tracking progress of this branch for several days on mainnet & testnet now so will merge this into main.

https://www.notion.so/pocketnetwork/Testing-Tendermint-fbcd3abbed5b4d1ea470a964e2e8b014?pvs=4
